packages/cli/core/src/commands/ui/deploy.ts

```diff
@@ -159,14 +160,47 @@ export default class Deploy extends CLICommand {
     this.validateDeployConfigJson(deployConfigPath, deployConfig);
     const hostedPage = this.buildHostedPage(deployConfig);
 
-    if (flags.pageId) {
-      await this.updatePage({...hostedPage, id: flags.pageId});
+    const pageId = flags.pageId || (await this.getPageIdFromPage(hostedPage));
+
+    if (pageId) {
+      await this.updatePage({...hostedPage, id: pageId});
       return;
     }
 
     await this.createPage(hostedPage);
   }
 
+  private async getPageIdFromPage(
+    hostedPage: New<HostedPage, null>
+  ): Promise<string | null> {
+    const {flags} = await this.parse(Deploy);
+    const orgId = getTargetOrg(this.configuration, flags.organization);
+    const client = await this.createClient();
+    const response = await client.hostedPages.list({
+      filter: hostedPage.name,
+      perPage: 1000,
```

Comment:
Is there a significance to the number 1000. Can it be extracted to a named constant to make its significance/purpose explicit?

Reply:
There's no significance, we could probably leave it by default tbh., given https://github.com/coveo/cli/pull/1234#discussion_r1151161482

Reply:
Fun fact, this is what made the E2E fails:
```
[11:02:45]APIError: 
[11:02:45]Status code: 400
[11:02:45]Error code: OutOfRangeParameterValueException
[11:02:45]Message: Value '1000' for parameter 'perPage' must be between 1 and 100
[11:02:45]Request ID: 0fe55e1f-388a-4580-9a4d-76bea6b78067
[11:02:45]    at wrapError (E:\Repos\cli\packages\cli\commons\src\errors\wrapError.ts:12:12)
[11:02:45]    at Deploy.catch (E:\Repos\cli\packages\cli\commons\src\command\cliCommand.ts:28:28)
[11:02:45]    at Deploy.trackError (E:\Repos\cli\packages\cli\commons\src\preconditions\trackable.ts:91:31)
[11:02:45]    at processTicksAndRejections (node:internal/process/task_queues:95:5)
[11:02:45]    at async Deploy._run (E:\Repos\cli\node_modules\@oclif\core\lib\command.js:84:13)
[11:02:45]    at async Config.runCommand (E:\Repos\cli\node_modules\@oclif\core\lib\config\config.js:272:25)
[11:02:45]    at async Object.run (E:\Repos\cli\node_modules\@oclif\core\lib\main.js:74:5)
```